drizzlepac/processInput.py

```diff
@@ -793,6 +794,10 @@ def changeSuffixinASN(asnfile, suffix):
     # Assign newly created, reformatted array to extension
     newasn = np.array(newdata,dtype=new_dtype)
     fasn[1].data = newasn
+    # Explicitly set the values of the boolean column to original values since they get
+    # lost when writing out as a FITS_Rec object using the FITS interface.
+    for row, dcol in zip(fasn[1].data, newdata):
```

Comment:
Will this work for older astropy versions as well?

Reply:
This change does not rely on any specific feature of astropy, as it simply copies data using numpy functionality directly.  This was confirmed by interactively processing data from 2 ASNs (both which were able to reproduce the original problem) without any problems using both astropy 4.3.post1 and 5.0.2.